python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -341,22 +351,61 @@ def _messages_thread(
         is_session_closed: Event,
     ) -> None:
         try:
+            # first, check if the target is readable. If it's now, we may need to receive extra params from the external process
+            if not self.can_start(params):
```

Comment:
Having nested `can_start` checks feels off to me. Why not remove this outer one and just wait for the launch payload?

Reply:
Because the starting condition is individual for each `PipesLogReader`. We can't assume they all will become startable at the same time. 

For example, they might be waiting for different objects to appear on S3, and thes objects are uploaded by an external system one by one.